libcudacxx/include/cuda/std/__format/format_integral.h

```diff
@@ -0,0 +1,237 @@
+    //                              ^ location of __first
+    // The zero padding is done like:
+    // - Write [sign][prefix]
+    // - Write data right aligned with '0' as fill character.
+    __out_it                  = _CUDA_VSTD::__fmt_copy(__array, __first, _CUDA_VSTD::move(__out_it));
+    __specs.__alignment_      = _CUDA_VSTD::to_underlying(__fmt_spec_alignment::__right);
+    __specs.__fill_.__data[0] = _CharT{'0'};
+    __specs.__width_ -= _CUDA_VSTD::min(static_cast<uint32_t>(__last - __first), __specs.__width_);
+  }
+
+  if (__specs.__std_.__type_ != __fmt_spec_type::__hexadecimal_upper_case)
+  {
+    return _CUDA_VSTD::__fmt_write(__first, __last, __ctx.out(), __specs);
+  }
+  return _CUDA_VSTD::__fmt_write_transformed(__first, __last, __ctx.out(), __specs, _CUDA_VSTD::__fmt_hex_to_upper);
+}
+
+template <class _Tp, class _CharT, class _FmtCtx>
+[[nodiscard]] _CCCL_API typename _FmtCtx::iterator
+__fmt_format_int(_Tp __value, _FmtCtx& __ctx, __fmt_parsed_spec<_CharT> __specs)
+{
+  static_assert(__cccl_is_integer_v<_Tp>);
+
+  const auto __uvalue   = ::cuda::uabs(__value);
+  const auto __negative = is_signed_v<_Tp> && __value < 0;
+
+  switch (__specs.__std_.__type_)
+  {
+    case __fmt_spec_type::__binary_lower_case: {
+      char __array[__fmt_int_buffer_size_v<_Tp, 2>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 2>, "0b", 2);
+    }
+    case __fmt_spec_type::__binary_upper_case: {
+      char __array[__fmt_int_buffer_size_v<_Tp, 2>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 2>, "0B", 2);
+    }
+    case __fmt_spec_type::__octal: {
+      // Octal is special; if __uvalue == 0 there's no prefix.
+      char __array[__fmt_int_buffer_size_v<_Tp, 8>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 8>, __uvalue != 0 ? "0" : nullptr, 8);
+    }
+    case __fmt_spec_type::__default:
+    case __fmt_spec_type::__decimal: {
+      char __array[__fmt_int_buffer_size_v<_Tp, 10>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 10>, nullptr, 10);
+    }
+    case __fmt_spec_type::__hexadecimal_lower_case: {
+      char __array[__fmt_int_buffer_size_v<_Tp, 16>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 16>, "0x", 16);
+    }
+    case __fmt_spec_type::__hexadecimal_upper_case: {
+      char __array[__fmt_int_buffer_size_v<_Tp, 16>];
+      return _CUDA_VSTD::__fmt_format_int_impl(
+        __uvalue, __ctx, __specs, __negative, __array, __fmt_int_buffer_size_v<_Tp, 16>, "0X", 16);
+    }
+    default:
+      _CCCL_UNREACHABLE();
+  }
+}
+
+template <class _CharT, class _FmtContext>
```

Comment:
I see that llvm puts bool into its own file which does make sense as it is completely different than integers